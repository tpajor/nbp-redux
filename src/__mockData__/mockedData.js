const card = { id: 'someId', code: 'USD', currency: 'dolar amerykański', rates: [{ ask: 3.3, bid: 3.4 }]};
const card2 = { id: 'someId2', code: 'USD', currency: 'dolar amerykański', rates: { ask: 3.3, bid: 3.4 }};

const search = {
  cards: [card, card2],
  temporaryCard: card,
  searchError: false,
  addCardError: true,
  addCardErrorMessage: 'error',
  currenciesTable: ['usd', 'chf', 'aud'],
  currencyCodeToViewInDetail: 'usd',
};

const user = { userName: 'User', pass: 'pass' };
const user2 = { userName: 'User2', pass: 'pass2'};

export const login = {
  signingError: true,
  signingErrorMessage: 'error',
  signedIn: true,
  userSignedIn: 'User',
  users: [user, user2]
};

const lastRates = [
  { ask: 3.5983, bid: 3.5271, effectiveDate: "2018-05-07", no: "087/C/NBP/2018" },
  { ask: 3.6088, bid: 3.5374, effectiveDate: "2018-05-08", no: "088/C/NBP/2018" }
];

const detail = {
  lastRates: lastRates,
  detailsInputError: false,
  detailsInputErrorMessage: '',
};

export const initialState = { search, login, detail };

export const searchViewData = {
  cards: search.cards,
  temporaryCard: search.temporaryCard,
  searchError: search.searchError,
  signedIn: login.signedIn,
  userSignedIn: login.userSignedIn,
  addCardError: search.addCardError,
  addCardErrorMessage: search.addCardErrorMessage,
  currenciesTable: search.currenciesTable,
};

export const detailViewData = {
  currencyCodeToViewInDetail: search.currencyCodeToViewInDetail,
  lastRates: detail.lastRates,
  detailsInputError: detail.detailsInputError,
  detailsInputErrorMessage: detail.detailsInputErrorMessage,
  signedIn: login.signedIn,
};