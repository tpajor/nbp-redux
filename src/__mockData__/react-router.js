const Link = () => {
  return jest.fn();
}

export default Link;