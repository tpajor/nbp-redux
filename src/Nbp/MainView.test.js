import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import MainViewConnected, { MainView } from './MainView';
import configureMockStore from 'redux-mock-store';
import { initialState, searchViewData, detailViewData } from '../__mockData__/mockedData'

const mockStore = configureMockStore();

const store = mockStore(initialState);

Enzyme.configure({ adapter: new Adapter() });

function setup() {
  const enzymeWrapper = shallow(<MainView store={store} />);

  return enzymeWrapper;
};

describe('MainView component', () => {
  it('should render self and subcomponents with their props', () => {
    const enzymeWrapper = setup();

    expect(enzymeWrapper.length).toBe(1);

    expect(enzymeWrapper.find('div').length).toBe(1);
    expect(enzymeWrapper.find('div').hasClass('row MainView')).toBe(true);

    enzymeWrapper.setProps({searchViewData, detailViewData});

    expect(enzymeWrapper.find('Connect(DetailContainer)').length).toBe(1);
    expect(enzymeWrapper.find('Connect(DetailContainer)').props().detailViewData).toEqual(detailViewData);
    
    expect(enzymeWrapper.find('Connect(Component)').length).toBe(1);
    expect(enzymeWrapper.find('Connect(Component)').props().searchViewData).toEqual(searchViewData);
  });

  it('should render correctly', () => {
    const enzymeWrapper = setup();

    expect(enzymeWrapper).toMatchSnapshot();
  });

  it('should map redux state to component props', () => {
    const enzymeWrapper = shallow(<MainViewConnected store={store} />);

    expect(enzymeWrapper.props().searchViewData).toEqual(searchViewData);
    expect(enzymeWrapper.props().detailViewData).toEqual(detailViewData);
  });
});