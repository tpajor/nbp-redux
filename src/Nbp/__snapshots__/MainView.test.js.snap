// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainView component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    store={
        Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(Component)
          searchViewData={undefined}
/>,
        <Connect(DetailContainer)
          detailViewData={undefined}
/>,
      ],
      "className": "row MainView",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "searchViewData": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "detailViewData": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(Component)
            searchViewData={undefined}
/>,
          <Connect(DetailContainer)
            detailViewData={undefined}
/>,
        ],
        "className": "row MainView",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "searchViewData": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "detailViewData": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
